cmd/maestro/server/grpc_server.go

```diff
@@ -133,11 +133,10 @@ func (svr *GRPCServer) Publish(ctx context.Context, pubReq *pbv1.PublishRequest)
 			if err != nil {
 				return nil, fmt.Errorf("failed to get resource: %v", err)
 			}
-			// handle the special case that the resource is updated by the source controller
-			// and the version of the resource in the request is less than it in the database
-			if found.Version < res.Version {
-				res.Version = found.Version
-			}
+			// keep the existing version for bundle resource, mainly from hub controller,
+			// the version is not guaranteed to be increased.
+			// TODO: handle the resource version for bundle resource from the source
+			res.Version = found.Version
```

Comment:
we may need compare the resource manifests instead of comparing version?

Reply:
update method of resource service will check the manifest equality, do we need to do it here?
https://github.com/openshift-online/maestro/blob/1e08a6a7cb942379ca04a49be4fe5a8616e122d6/pkg/services/resource.go#L102-L104